tools/python/create_reduced_build_config.py

```diff
@@ -0,0 +1,129 @@
+                _extract_ops_from_onnx_graph(attr.g, operators, domain_opset_map)
+
+
+def _process_onnx_model(model_path, required_ops):
+    model = onnx.load(model_path)
+
+    # create map of domain to opset for the model
+    domain_opset_map = {}
+    for opset in model.opset_import:
+        # empty domain == ai.onnx
+        domain = opset.domain if opset.domain else 'ai.onnx'
+        domain_opset_map[domain] = opset.version
+
+        if domain not in required_ops:
+            required_ops[domain] = {opset.version: set()}
+        elif opset.version not in required_ops[domain]:
+            required_ops[domain][opset.version] = set()
+
+    # check the model imports at least one opset. if it does not it's an unexpected edge case that we have to ignore
+    # as we don't know what opset nodes in the graph belong to.
+    if domain_opset_map:
+        _extract_ops_from_onnx_graph(model.graph, required_ops, domain_opset_map)
+
+
+def _extract_ops_from_onnx_model(model_path_or_dir):
+    '''Extract ops from a single ONNX model, or all ONNX models found by recursing model_path_or_dir'''
+
+    if not os.path.exists(model_path_or_dir):
+        raise ValueError('Path to model/s does not exist: {}'.format(model_path_or_dir))
+
+    required_ops = {}
+
+    if os.path.isfile(model_path_or_dir):
+        _process_onnx_model(model_path_or_dir, required_ops)
+    else:
+        for root, _, files in os.walk(model_path_or_dir):
+            for file in files:
+                if file.lower().endswith('.onnx'):
+                    model_path = os.path.join(root, file)
+                    _process_onnx_model(model_path, required_ops)
+
+    return required_ops
+
+
+def create_config_from_onnx_models(model_path_or_dir: str, output_file: str):
+
+    required_ops = _extract_ops_from_onnx_model(model_path_or_dir)
+
+    directory, filename = os.path.split(output_file)
+    if not filename:
+        raise RuntimeError("Invalid output path for configuation: {}".format(output_file))
+
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+    with open(output_file, 'w') as out:
+        out.write("# Generated from ONNX models path of {}\n".format(model_path_or_dir))
+
+        for domain in sorted(required_ops.keys()):
+            for opset in sorted(required_ops[domain].keys()):
+                ops = required_ops[domain][opset]
+                if ops:
+                    out.write("{};{};{}\n".format(domain, opset, ','.join(sorted(ops))))
+
+
+def main():
+    argparser = argparse.ArgumentParser(
+        'Script to create a reduced build config file from either ONNX or ORT format model/s. '
+        'See /docs/Reduced_Operator_Kernel_build.md for more information on the configuration file format.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    argparser.add_argument('-f', '--format', choices=['ONNX', 'ORT'], default='ONNX',
+                           help='Format of model/s to process.')
```

Comment:
Could there be any need to create config on a folder with both onnx and ort files?  Since full ORT can run ort model also.

Reply:
If that is needed you can create two configs and concat them together. 

The general internal usage by the conversion script is to create the .ort format models in the same dir as the original models, but as we potentially do some optimization when creating the .ort model the ops won't necessarily match the onnx model. Due to that we don't want to do both.

Also can't mix the two if type reduction is enabled, as we wouldn't get the type info from the onnx models, leading to potentially incomplete lists for the required types. 

If a real use case comes up where it's actually needed we can reconsider whether we should support that.